packages/foam-vscode/src/dated-notes.test.ts

```diff
@@ -0,0 +1,47 @@
+import { workspace } from 'vscode';
+import { getDailyNotePath } from './dated-notes';
+import skipIf from 'skip-if';
+
+const skipIfWindows = skipIf(process.platform === 'win32');
+const skipIfNotWindows = skipIf(process.platform !== 'win32');
+
+describe('getDailyNotePath', () => {
+  test('Adds the root directory to relative directories', async () => {
+    const date = new Date('2021-02-07T00:00:00Z');
+    await workspace
+      .getConfiguration('foam')
+      .update('openDailyNote.directory', 'journal');
+    const foamConfiguration = workspace.getConfiguration('foam');
+    expect(getDailyNotePath(foamConfiguration, date)).toMatch(
+      new RegExp(/journal[\\/]2021-02-07.md$/)
+    );
+  });
+
+  skipIfWindows(
+    'Uses absolute directories without modification (Unix systems)',
+    async () => {
+      const date = new Date('2021-02-07T00:00:00Z');
+      await workspace
+        .getConfiguration('foam')
+        .update('openDailyNote.directory', '/absolute_path/journal');
+      const foamConfiguration = workspace.getConfiguration('foam');
+      expect(getDailyNotePath(foamConfiguration, date)).toMatch(
```

Comment:
iiuc you are doing all this to deal with the different path format.

I understand the issue, that why we have started using `URI` across the code, except for the lowest fs service level (basically all platform specific translation is managed by the `FileDataStore`.

I would suggest you change `getDailyNotePath` to use URI instead, and consolidate the test by accessing the `path` property on the URI (which is not platform specific).

Beside making the tests easier to read/follow, and removing the skip dependency, this will make the code more solid in the first place.

thoughts?

Reply:
That seems to make sense. This is actually my first time writing Typescript, so I didn't know about `URI` 😅 .

I'll take a look and rework this PR later this week. Thanks for the great feedback! 🎉

Reply:
hehe JS/TS are weird languages in many ways, welcome to the club :)

`URI` is actually not built into JS/TS, it's a utility class (that we shamelessly copied from the vscode repo to maintain compatibility) that we use exactly to abstract the platform specific details.
As it's compatible with vscode's, here you can decide whether to use our `URI` (see `foam-core/src/common/uri.ts` or the one coming from vscode (with a slight preference for our implementation for consistency))

Reply:
I've reworked it to use `foam-core`'s `URI` in _more_ places. Pushing the use of URI up a level allows the tests to avoid using the `skip-if` dependency.